### ISSUE TITLE
Warning under DMCA

### ISSUE THREAD

**Issue by kyranet** [fnbrjs/fnbr.js#188]
Hello, I was passing by and noticed that there is a file which is a copy-paste of [async-queue](https://github.com/sapphiredev/utilities/blob/ba8183287dbbc3f3d7d79af6d5a2d3dd8d62f63e/packages/async-queue/src/lib/AsyncQueue.ts) with slightly different names and identical logic. The offending code can be seen below:

https://github.com/fnbrjs/fnbr.js/blob/7b83474a071d7a0edda07411311891f774242bd7/src/util/AsyncQueue.ts#L6-L50

Which is identical to the file of the aforementioned original source code.

Furthermore, this repository does not comply with the LICENSE requirements from `sapphiredev/utilities`, which `async-queue` inherits due to being part of it. This license in question is **The MIT License**, and can be seen [here](https://github.com/sapphiredev/utilities/blob/main/LICENSE.md).

Under the MIT license, you have some conditions for the re-distribution of the work, which is to keep the **license and copyright notice**. However, none can be seen in this repository as of 7b83474a071d7a0edda07411311891f774242bd7, which is the last commit made to the moment of writing this issue.

You may resolve this issue by pasting a full copy of our aforementioned license and a copyright notice within the next 7 days. You can find a valid copyright notice below:

```javascript
// MIT License | Copyright © `2020` `The Sapphire Community and its contributors`
```

---

As a side note, the original code is published under `@sapphire/async-queue` in npm, so you do not need to keep a copy of the software here.

**Comment by ThisNils** [comment]
Alright, I added the copyright notice you suggested.
https://github.com/fnbrjs/fnbr.js/blob/0f94e5988131a3c448870226a195ea87c998bf1e/src/util/AsyncQueue.ts#L1

**Comment by thehairy** [comment]
> Alright, I added the copyright notice you suggested.
> 
> https://github.com/fnbrjs/fnbr.js/blob/0f94e5988131a3c448870226a195ea87c998bf1e/src/util/AsyncQueue.ts#L1

Don't forget a copy of the license :)

**Comment by kyranet** [comment]
You can create a `THIRD_PARTY_LICENSES` file and paste the license there. As long as it's distributed with the code it's fine, it doesn't have to be in the files licensed under it, it just makes re-distribution easier.

And yeah, you have to, it's the MIT license's first part of its only condition.

As a side note, is there a reason why you don't want to use the module from npm? It would save you some of the legal hassle.

**Comment by kyranet** [comment]
Looks resolved in 029fb3f2df4f3cb7b5acc9b944246233d0a2b960, thank you :)

**Comment by ThisNils** [comment]
I added the full license now:
https://github.com/fnbrjs/fnbr.js/blob/20eb01a5d1cc23d8bfd54e84b32c3ec6c04230cc/src/util/AsyncQueue.ts#L1-L9

I'll probably use the module directly from npm at some point, for now I'll keep it this way.
